### PR TITLE
docs: add git workflow [v2]

### PR DESCRIPTION
**Pull request by apahim**
v2:
- Incorporated all suggestions from v1 (#1435) review.

v1: #1435 
- Create a git workflow section based on the devel list
discussion.

### CODE REVIEW

File: docs/source/ContributionGuide.rst [hunk 1]
```diff
@@ -65,10 +65,134 @@ be submitted as a github Pull Request. This process tries to ensure that every
...
+
+    $ git clone git@github.com:<username>/avocado.git
+
+- Enter the directory::
+
+    $ cd avocado
+
+- Create a ``remote``, pointing to the upstream::
+
+    $ git remote add upstream git@github.com:avocado-framework/avocado-misc-tests.git
```
Comment (clebergnu) [inline docs/source/ContributionGuide.rst]: I guess you intended to point to `git@github.com:avocado-framework/avocado.git`, right?

Comment (apahim) [inline docs/source/ContributionGuide.rst]: Yes :)

Comment (clebergnu) [inline docs/source/ContributionGuide.rst]: Please fix this little issue, and I can merge right away.

**Comment by apahim** [comment]
@clebergnu, updated.

**Pull request merged**